scala-thoughts.md

```diff
@@ -0,0 +1,5 @@
+In theory, I think, it's pretty easy to do in Scala anything that's easy to do in Java. Some of that may not be "good" Scala, though. Basic object-oriented programming, for example isn't any more difficult (even if the idea of companion objects is kind of weird at first). Functional programming is pretty easy in Scala as well, although I was surprised to find that higher-order functions are typically methods of types for which those functions make any sense. I found that it was
+quite difficult to use recursion and pattern-matching to do something better handled without the pattern-matching. In particular, I basically gave up figuring out how to implement zipMultiple with pattern-matching, looked at the sample solution, realized _why_ I hadn't figured it out (it was considerably more complicated than I think is necessary), and ended up just referring to Prof. Ben's solution. Obviously there are going to be some tasks that pattern matching won't be suited for regardless of the
+language, but I'm not sure whether this is one of them, or if Scala's pattern matching is particularly not suited to it.
+
```

Comment:
I mostly wish there were a combined version of Python (which is still my favorite) and Scala (although that might be _the_ slowest programming language out there if it existed). Scala has a lot of really cool features that Python doesn't have at all, and I think Scala does those really well. On the other hand, I prefer Python's syntax and also I tend to find that it behaves more intuitively for me (tuples actually being iterable, for one thing...).